src/algebra/homology/homotopy_category.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.homology.homotopy
+import category_theory.quotient
+
+/-!
+# The homotopy category
+
+`homotopy_category V c` gives the category of chain complexes of shape `c` in `V`,
+with chain maps identified when they are homotopic.
+-/
+
+universes v u
+
+open_locale classical
+noncomputable theory
+
+open category_theory category_theory.limits homological_complex
+
+variables {ι : Type*}
+variables (V : Type u) [category.{v} V] [preadditive V]
+variables (c : complex_shape ι)
+
+/--
+The congruence on `homological_complex V c` given by the existence of a homotopy.
+-/
+def homotopic : hom_rel (homological_complex V c) := λ C D f g, nonempty (homotopy f g)
+
+instance homotopy_congruence :
+  congruence (homotopic V c) :=
+{ is_equiv := λ C D,
+  { refl := λ C, ⟨homotopy.refl C⟩,
+    symm := λ f g ⟨w⟩, ⟨w.symm⟩,
+    trans := λ f g h ⟨w₁⟩ ⟨w₂⟩, ⟨w₁.trans w₂⟩, },
+  comp_left := λ E F G m₁ m₂ g ⟨i⟩, ⟨i.comp_left _⟩,
+  comp_right := λ E F G f m₁ m₂ ⟨i⟩, ⟨i.comp_right _⟩, }
+
+/--
+`homotopy_category V c` is the category of chain complexes of shape `c` in `V`,
+with chain maps identified when they are homotopic.
+-/
+@[derive category]
+def homotopy_category :=
+category_theory.quotient (homotopic V c)
```

Comment:
I personally wouldn't mind putting a bit more on one line. But it's an extremely minor comment.
```suggestion
/-- `homotopy_category V c` is the category of chain complexes of shape `c` in `V`,
with chain maps identified when they are homotopic. -/
@[derive category]
def homotopy_category := category_theory.quotient (homotopic V c)
```